tuya.cpp

```diff
@@ -136,12 +138,13 @@ void DeRestPluginPrivate::handleTuyaClusterIndication(const deCONZ::ApsDataIndic
         uint8_t status;
         uint8_t transid;
         int16_t dp;
-        uint8_t dp_type;
-        uint8_t dp_identifier;
         uint8_t fn;
+        quint8 length = 0;
         qint32 data = 0;
+        
+        uint8_t dp_type;
+        uint8_t dp_identifier;
```

Comment:
I still don't know what `dp` means, data pointer? Please use a more descriptive variable name :)

Please always use camelCase and Qt data types for consistency.

`quint8` instead `uint8_t`, ...

`quint8 dpType;`

Reply:
It s name from the first person that have sniffed the tuya protocol https://medium.com/@dzegarra/zigbee2mqtt-how-to-add-support-for-a-new-tuya-based-device-part-2-5492707e882d

![img](https://miro.medium.com/max/657/1*moHdm0bw7PZLui2NkZT2iA.png)

I think they are used on all zigbee project ATM

Reply:
Aha ok, the table in the link is quite useful. I would propose to give it a small comment with description.

```
// "dp" field describes the action/message of a command frame
// https://medium.com/@dzegarra/zigbee2mqtt-how-to-add-support-for-a-new-tuya-based-device-part-2-5492707e882d

quint16 dp;
```

Shouldn't dp be uint16 instead of int16?

Reply:
The link is at line 100, will add the comment for dp
And yes dp is already corrected 

>quint16 dp;

was using int16_t previously but the value is unsigned.

Reply:
Done